File: Backend/Martins.Conciliacao.API/Controllers/PeriodoController.vb
Imports System.Web.Http
Imports Martins.Conciliacao.Code.Periodo
Imports Martins.Conciliacao.Model
Imports Microsoft.Owin.Security.OAuth

Namespace Controllers
  Public Class PeriodoController
    Inherits ApiController
    Private context As OAuthGrantResourceOwnerCredentialsContext


    Public Sub PeriodoController(contexto As OAuthGrantResourceOwnerCredentialsContext)
      context = contexto
    End Sub
    ' GET api/<controller>
    Public Function [Get](nome As String, situacao As String) As IEnumerable(Of PeriodoModel)
      Dim bll As New PeriodoBLL()
      If nome IsNot Nothing Then
        nome = "%" & nome & "%"
      End If
      Return bll.selectPeriodo(nome, situacao)
    End Function
    ' GET api/<controller>
    Public Function [Get]() As IEnumerable(Of PeriodoModel)
      Dim bll As New PeriodoBLL()
      Return bll.selectPeriodo(Nothing, Nothing)
    End Function
    ' GET api/<controller>
    Public Function [Get](nome As String) As IEnumerable(Of PeriodoModel)
      Dim bll As New PeriodoBLL()
      If nome IsNot Nothing Then
        nome = "%" & nome & "%"
      End If
      Return bll.selectPeriodo(nome, Nothing)
    End Function
    ' GET api/<controller>/5
    Public Function [Get](id As Integer) As PeriodoModel
      Dim bll As New PeriodoBLL()

      Return bll.selectPeriodoPorId(id)
    End Function

    ' POST api/<controller>
    Public Sub [Post](<FromBody> periodo As PeriodoModel)
      Dim bll As New PeriodoBLL()
      If bll.selectPeriodo(periodo.nome, Nothing).Count > 0 Then
        Throw New Exception("Nome do Período já existente")
        Exit Sub
      End If
      bll.insertPeriodo(periodo)
    End Sub

    ' PUT api/<controller>/5
    Public Sub Put(<FromBody> periodo As PeriodoModel, acao As String)
      Dim bll As New PeriodoBLL()
      If acao.ToLower().Equals("ativar") OrElse acao.ToLower().Equals("inativar") Then
        bll.ativarDesativarPeriodo(periodo)
      Else
        bll.updatePeriodo(periodo)
      End If
    End Sub

    ' DELETE api/<controller>/5
    Public Sub Delete(id As Integer)
      Dim bll As New PeriodoBLL()
      bll.deletePeriodo(id)
    End Sub
  End Class
End Namespace

File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Controllers/PeriodoController.cs
using JsonWebToken.Code;
using JsonWebToken.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace JsonWebToken.Controllers
{
  public class PeriodoController : ApiController
  {
    // GET api/<controller>
    public IEnumerable<PeriodoModel> Get(string nome, string situacao)
    {
      PeriodoBLL bll = new PeriodoBLL();
      if (nome != null) {
        nome = "%"+ nome + "%";
      }
      return bll.selectPeriodo(nome,situacao);
    }
    // GET api/<controller>
    public IEnumerable<PeriodoModel> Get()
    {
      PeriodoBLL bll = new PeriodoBLL();
      
      return bll.selectPeriodo(null,null);
    }
    // GET api/<controller>
    public IEnumerable<PeriodoModel> Get(string nome)
    {
      PeriodoBLL bll = new PeriodoBLL();
      if (nome != null)
      {
        nome = "%" + nome + "%";
      }
      return bll.selectPeriodo(nome, null);
    }
    // GET api/<controller>/5
    public PeriodoModel Get(int id)
    {
      PeriodoBLL bll = new PeriodoBLL();

      return bll.selectPeriodoPorId(id);
    }

    // POST api/<controller>
    public void Post([FromBody]PeriodoModel periodo)
    {
      PeriodoBLL bll = new PeriodoBLL();
      if (bll.selectPeriodo(periodo.nome, null).Count > 0)
      {
        throw new Exception("Nome do Período já existente");
      }
      bll.insertPeriodo(periodo);
    }

    // PUT api/<controller>/5
    public void Put( [FromBody]PeriodoModel periodo, string acao)
    {
      PeriodoBLL bll = new PeriodoBLL();
      if (acao.ToLower().Equals("ativar")|| acao.ToLower().Equals("desativar"))
      {
        bll.ativarDesativarPeriodo(periodo);
      } else 
      {
        bll.updatePeriodo(periodo);
      }
    }

    // DELETE api/<controller>/5
    public void Delete(int id)
    {
      PeriodoBLL bll = new PeriodoBLL();
      bll.deletePeriodo(id);
    }
  }
}
